Add transactional batch sample and clean up others (#18065)
Azure/azure-sdk-for-net: sdk/tables/Azure.Data.Tables/samples/Sample5UpdateUpsertEntities.md

# Updating and Upserting Table Entities

> This sample demonstrates how to query a table for entities. You will have needed to [create a table](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/samples/Sample1CreateDeleteTables.md) in the service in order to query entities from it. To get started, you'll need access to either a Storage or Cosmos DB account.

Reviewer: > You will have needed to ...

That reads a bit awkward to me.  Maybe something like:   `You will need to have previously created a table..."  ?

## Create a `TableClient`

A `TableClient` is needed to perform table-level operations like inserting and deleting entities within the table, so it is ideal for dealing with only a specific table. There are two ways to get a `TableClient`:
- Call `GetTableClient` from the `TableServiceClient` with the table name.

```C# Snippet:TablesSample1GetTableClient

> tableName = "OfficeSupplies1p2";

Reviewer: nit:  This looks strange without the `var` or type name...

var tableClient = serviceClient.GetTableClient(tableName);
```

- Create a `TableClient` with a SAS URI, an endpoint and `TableSharedKeyCredential`, or a connection string.

```C# Snippet:TablesSample1CreateTableClient

> tableClient = new TableClient(

Reviewer: nit:  This looks strange without the `var` or type name...

    new Uri(storageUri),

>     tableName,

Reviewer: nit:  Should this be defined in the snippet so that it's self-contained?

    new TableSharedKeyCredential(accountName, storageAccountKey));
```

If you are not familiar with creating tables, refer to the sample on [creating and deleting tables](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/samples/Sample1CreateDeleteTables.md).

## Upsert an entity

> The upsert operation combines the capabilities of insert and update unconditionally.

Reviewer: You may want to just touch on the semantics of an upsert for those unfamiliar... or maybe link to it somewhere?

Author: I debated it, but the snippet comments touch on them. Do you think it makes sense to expand on it more here also?

Reviewer 2: I had no idea what an upsert was but the text gave me enough context to understand.

Reviewer: I'll defer to your judgement;  I'm tainted by knowing what it is and trying to project "what would I think if I didn't...."  so I may be a poor perspective.

```C# Snippet:TablesSample5UpsertEntityAsync
var entity = new TableEntity(partitionKey, rowKey)
{
    {"Product", "Markers" },
    {"Price", 5.00 },
    {"Brand", "myCompany" }
};

> // Entity doesn't exist in table, so invoking UpsertEntity will simply insert the entity.

Reviewer: ```suggestion
// The entity doesn't exist in table, so invoking UpsertEntity will insert a new entity.
```

await tableClient.UpsertEntityAsync(entity);
```

```C# Snippet:TablesSample5UpsertWithReplaceAsync
// Delete an entity property.
entity.Remove("Brand");

> // Entity does exist in the table, so invoking UpsertEntity will update using the given UpdateMode, which defaults to Merge if not given.

Reviewer: ```suggestion
// The entity exists in the table, so invoking UpsertEntity will update the entity.  It will apply the configured UpdateMode, which defaults to Merge if not specified.
```
